File: c class/stack.c
#include<stdio.h>
#define size 5
int top=-1,stack[size];
void push(int x){
    if(top==size-1){
        printf("stack is full");
    }
    stack[++top]=x;
   
}
int pop(){
    if(top==-1){
        printf("stack is empty");
    }
    return(stack[top--]);
}
void traverse(){
    int i;
    printf("\n");
    for(i=top;i>=0;i--){
        printf("%d \t ",stack[i]);
    }
    
} 
int main(){
    push(3);push(2);push(8);
    traverse();
    pop();traverse();
    push(33);push(1);pop();traverse();
}

File: java/stack/stack.java
package stack;
public class stack{
    int stack[];
    int top=-1;
    int capacity;

    public stack(int size){
        stack=new int[size];
        top=-1;
        capacity=size;
    }
    public void push(int data){
        if(top==capacity-1){
            System.out.println("Stack is full");
        }
        else{
            stack[++top]=data;
            System.out.println("Data inserted: ");
        }
    }
    public void pop(){
        if(top==-1){
            System.out.println("Stack is empty");
        }
        else{
            top--;
        }
    }
    public void display(){
        if(top==-1){
            System.out.println("Stack empty");
        }
        else{
        System.out.println("Stack elements:");
        for(int i=0;i<=top;i++){
            System.out.println(stack[i]);
        }
        }
    }
}
